src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -145,30 +145,41 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             //      if (s != null) { ... }
             //
             // It's no longer safe to use pattern-matching because 'field is string s' would never be true.
+            // 
+            // Additionally, also bail out if the assigned local is referenced (read or write) up to the point of null check.
+            //      var s = field as string;
+            //      MethodCall(flag: s == null);
+            //      if (s != null) { ... }
```

Comment:
your comment doesn't seem to match. it says "(read or write)", but below you only do `identifierName.IsWrittenTo()`.  Intentional?

Reply:
`identifierName.IsWrittenTo()` check is for the asOperand, which was already done prior to this change. My change only adds the if statement right below that if statement.